Bottom: 196cede6baad3bc51a906329b3a4220ef1b40be9
Top:    f7bc05d643549e8fd10f6f182184d5905afa16ca
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 06:58:11 +0100

Refresh of hid-gtk-add-consolidated-api-f

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 6a37703..11c0193 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -32,127 +32,6 @@ RCSID ("$Id$");
 
 bool ghid_flip_x = false, ghid_flip_y = false;
 
-static void ghid_zoom_view_fit (void);
-
-static void
-pan_common (GHidPort *port)
-{
-  int event_x, event_y;
-
-  /* We need to fix up the PCB coordinates corresponding to the last
-   * event so convert it back to event coordinates temporarily. */
-//  ghid_pcb_to_event_coords (gport->pcb_x, gport->pcb_y, &event_x, &event_y);
-
-  /* Don't pan so far the board is completely off the screen */
-  port->view_x0 = MAX (-port->view_width,  port->view_x0);
-  port->view_y0 = MAX (-port->view_height, port->view_y0);
-  port->view_x0 = MIN ( port->view_x0, PCB->MaxWidth);
-  port->view_y0 = MIN ( port->view_y0, PCB->MaxHeight);
-
-  ghid_note_event_location (NULL);
-
-  /* Fix up noted event coordinates to match where we clamped */
-//  ghid_event_to_pcb_coords (event_x, event_y, &gport->pcb_x, &gport->pcb_y);
-
-  ghidgui->adjustment_changed_holdoff = TRUE;
-  gtk_range_set_value (GTK_RANGE (ghidgui->h_range), gport->view_x0);
-  gtk_range_set_value (GTK_RANGE (ghidgui->v_range), gport->view_y0);
-  ghidgui->adjustment_changed_holdoff = FALSE;
-
-  ghid_port_ranges_changed();
-}
-
-static void
-ghid_pan_view_abs (Coord pcb_x, Coord pcb_y, int widget_x, int widget_y)
-{
-  gport->view_x0 = SIDE_X (pcb_x) - widget_x * gport->zoom;
-  gport->view_y0 = SIDE_Y (pcb_y) - widget_y * gport->zoom;
-
-  pan_common (gport);
-}
-
-void
-ghid_pan_view_rel (Coord dx, Coord dy)
-{
-  gport->view_x0 += dx;
-  gport->view_y0 += dy;
-
-  pan_common (gport);
-}
-
-
-/* gport->zoom:
- * zoom value is PCB units per screen pixel.  Larger numbers mean zooming
- * out - the largest value means you are looking at the whole board.
- *
- * gport->view_width and gport->view_height are in PCB coordinates
- */
-
-#define ALLOW_ZOOM_OUT_BY 5
-static void
-ghid_zoom_view_abs (Coord center_x, Coord center_y, double new_zoom)
-{
-  double min_zoom, max_zoom;
-  double xtmp, ytmp;
-
-  /* Limit the "minimum" zoom constant (maximum zoom), at 1 pixel per PCB
-   * unit, and set the "maximum" zoom constant (minimum zoom), such that
-   * the entire board just fits inside the viewport
-   */
-  min_zoom = 1;
-  max_zoom = MAX (PCB->MaxWidth  / gport->width,
-                  PCB->MaxHeight / gport->height) * ALLOW_ZOOM_OUT_BY;
-  new_zoom = MIN (MAX (min_zoom, new_zoom), max_zoom);
-
-  if (gport->zoom == new_zoom)
-    return;
-
-  xtmp = (SIDE_X (center_x) - gport->view_x0) / (double)gport->view_width;
-  ytmp = (SIDE_Y (center_y) - gport->view_y0) / (double)gport->view_height;
-
-  gport->zoom = new_zoom;
-  pixel_slop = new_zoom;
-  ghid_port_ranges_scale ();
-
-  gport->view_x0 = SIDE_X (center_x) - xtmp * gport->view_width;
-  gport->view_y0 = SIDE_Y (center_y) - ytmp * gport->view_height;
-
-  pan_common (gport);
-
-  ghid_set_status_line_label ();
-}
-
-static void
-ghid_zoom_view_rel (Coord center_x, Coord center_y, double factor)
-{
-  ghid_zoom_view_abs (center_x, center_y, gport->zoom * factor);
-}
-
-static void
-ghid_zoom_view_fit (void)
-{
-  ghid_pan_view_abs (0, 0, 0, 0);
-  ghid_zoom_view_abs (0, 0, MAX (PCB->MaxWidth  / gport->width,
-                                 PCB->MaxHeight / gport->height));
-}
-
-static void
-ghid_flip_view (Coord center_x, Coord center_y, bool flip_x, bool flip_y)
-{
-  int widget_x, widget_y;
-
-  /* Work out where on the screen the flip point is */
-  ghid_pcb_to_event_coords (center_x, center_y, &widget_x, &widget_y);
-
-  ghid_flip_x = ghid_flip_x != flip_x;
-  ghid_flip_y = ghid_flip_y != flip_y;
-
-  /* Pan the board so the center location remains in the same place */
-  ghid_pan_view_abs (center_x, center_y, widget_x, widget_y);
-
-  ghid_invalidate_all ();
-}
-
 /* ------------------------------------------------------------ */
 
 static const char zoom_syntax[] =
